src/charm.py

```diff
@@ -332,32 +407,40 @@ def _get_context_from_relations(self) -> dict:
                 if len(rel.units) > 0:
                     # We want to always pick the same unit, so sort the set
                     # before picking the first one.
-                    u = sorted(rel.units, key=lambda x: x.name)[0]
+                    unit = sorted(rel.units, key=lambda x: x.name)[0]
 
                     if len(rel.units) > 1:
                         logger.warning(
                             'Multiple units detected in the relation "%s:%s", '
                             "using only the first one (id: %s) for relation data.",
                             rel.name,
                             rel.id,
-                            u.name,
+                            unit.name,
                         )
                     if rel.name not in ctx:  # can be present from the "special" relations above
                         ctx[rel.name] = {}
-                    for k, v in rel.data[u].items():
-                        ctx[rel.name][k] = v
+                    for key, value in rel.data[unit].items():
+                        ctx[rel.name][key] = value
 
         return ctx
 
-    def _validate_yaml(self, supposed_yaml: str, expected_type: type) -> None:
-        """Validate that the supplied YAML is parsed into the supplied type."""
+    def _validate_yaml(self, supposed_yaml: str, expected_type: type) -> Union[bool, None]:
+        """Validate that the supplied YAML is parsed into the supplied type.
+
+        Args:
+            supposed_yaml: Yaml content to validate.
+            expected_type: Expected yaml type.
+
+        Returns:
+            An error if the validation goes wrong.
```

Comment:
"he validation goes wrong" seems too generic? How about "if the yaml is incorrect"?